### PR TITLE
Fix react-query cache persistence

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Found a bug where we end up always clearing the query cache on page load. This adjusts some of the logic with where/how we reset that cache so we benefit from loading up the cache from IDB. 

Instead of the logic that was there, which would clear the cache anytime the user id changes in `UserContext`, I've adjust it to only do that when we have a user in context already. This avoids clearing it on initial page load. I've also added explicit clearing of the cache on logout and login.

I've also updated the icons in the repository selector to remove the color distinction between projects non-project suggestions. We may revisit the iconography and colors later.

image

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 1e0af49</samp>

This pull request enhances the data fetching and caching logic for the login and user menu components, using React Query hooks and clearing the cache when appropriate. It also improves the user context management by syncing it with the query cache.

</details>

## Related Issue(s)

Fixes EXP-655

## How to test

* Visit the `/new` page for creating a new workspace.
* The first time you visit it, the repository selector should have a loading state until it finishes loading the suggestions.
* Do a hard refresh and it should show suggestions right away w/o a loading state.
* Verify projects are listed in the selector w/o any color indicator.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-queryfc19f41826</li>
	<li><b>🔗 URL</b> - <a href="https://brad-queryfc19f41826.preview.gitpod-dev.com/workspaces" target="_blank">brad-queryfc19f41826.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - brad-query-cache-get-suggested-repos-gha.17339</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-brad-queryfc19f41826%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by gtsiolis — APPROVED** [review]
Preview environment is not working but UX LGTM in the screenshot. 🏁 
Thanks @selfcontained. 🌮 

I kinda miss the differentiation of the project, but hopefully we can get back to it soon.

**Review by svenefftinge — APPROVED** [review]

File: components/dashboard/src/user-context.tsx [hunk 1]
```diff
@@ -29,7 +29,9 @@ const UserContextProvider: React.FC = ({ children }) => {
     const doSetUser = useCallback(
         (updatedUser: User) => {
             updateErrorUserDetails(updatedUser);
-            if (user?.id !== updatedUser.id) {
+            // If user has changed clear cache
+            // Ignore the case where user hasn't been set yet - initial load
```
Comment (AlexTugarev) [inline components/dashboard/src/user-context.tsx:33]: 🥇

**Pull request merged**